internal/tokens/oauth.go

```diff
@@ -66,19 +67,44 @@ func combinedClientInfoHandler(r *http.Request) (clientID, clientSecret string,
 	return
 }
 
+func isUriValid(clientHost, redirectHost string) bool {
+	clientHostParts := strings.Split(clientHost, ".")
+	redirectHostParts := strings.Split(redirectHost, ".")
+
+	if len(clientHostParts) != len(redirectHostParts) {
+		return false
+	}
+
+	for i := range clientHostParts {
+		if clientHostParts[i] == "*" {
+			continue
+		}
+		if redirectHostParts[i] != clientHostParts[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
 func checkRedirectUriDomain(baseURI, redirectURI string) error {
 	parsedClientUri, err := url.Parse(baseURI)
 	if err != nil {
-		return err
+			return err
 	}
 	parsedRedirect, err := url.Parse(redirectURI)
 	if err != nil {
-		return err
+			return err
 	}
-	if parsedClientUri.Host != parsedRedirect.Host || parsedClientUri.Scheme != parsedRedirect.Scheme {
-		err = fmt.Errorf("Wrong Redirect URI. Provided: [ Scheme: %s, Host: %s ], Expected: [ Scheme: %s, Host: %s ]", parsedRedirect.Scheme, parsedRedirect.Host, parsedClientUri.Scheme, parsedClientUri.Host)
-		sentry.CaptureException(err)
-		return err
+
+	clientHost := parsedClientUri.Host
+	redirectHost := parsedRedirect.Host
+
+	if parsedClientUri.Scheme == parsedRedirect.Scheme && isUriValid(clientHost, redirectHost) {
```

Comment:
normally we check errors first and return the success case at the bottom (see the diff for how it was previously done)